public/js/graph/react_node.js.jsx

```diff
@@ -0,0 +1,109 @@
+function renderReactGraph() {
+    React.render(
+        <ReactSVG width={1195} height={650}/>,
+        document.getElementById('react-graph')
+    );
+}
+
+function getAttributes(svgAttributes) {
+    var attrs = [];
+    //Traversing a NodeNamedMap type
+    Array.prototype.slice.call(svgAttributes).forEach(function(item) {
+        attrs[item.name] = item.value;
+    });
+    return attrs;
+}
+
+function getStyles(styles_strings) {
+    var styles = {};
+    if (!styles_strings) {
+        return styles;
+    }
+    styles_strings.split(";").map(function(key, value){
+        if (key) {
+            styles[key.substring(0, key.indexOf(':'))] = key.substring(key.indexOf(':')+1);
+        }
+    });
+    return styles;
+}
+
+var ReactSVG = React.createClass({
+    render: function() {
+        return (
+            <svg width={this.props.width} height={this.props.height}>
+                {/*<ReactRegions/>*/}
+                <ReactNodes/>
+                {/*<ReactRegions/>*/}
+                {/*<ReactBools/>*/}
+                {/*<ReactEdges/>*/}
+                {/*<ReactRegionLabels/>*/}
+            </svg>
+        );
+    }
+});
+
+var ReactNodes = React.createClass({
+    getInitialState: function(){
+        return {
+            nodes_list: []
+        };
+    },
+    
+    componentDidMount: function(){
+        var dict_list = [];
+        
+        $('.node').map(function(key, value) {
+            var entry = {};
+            entry["id"] = value.id;
+            entry["attributes"] = getAttributes(value.attributes);
+            //assume no styles here
+            entry["style"] = getStyles(entry["attributes"]["style"]);
+            entry["children"] = [];
+            //value.children is an HTML collection, converting to array here
+            var children_arr = [].slice.call(value.children);
+            //assume only one level of children
+            //for each children
+            children_arr.map(function(child) {
+                var child_entry = {};
+                child_entry["attributes"] = getAttributes(child.attributes);
+                child_entry["style"] = getStyles(child_entry["attributes"]["style"]);
+                //innerHTML is just for the text tag
+                child_entry["innerHTML"] = child.innerHTML;
```

Comment:
I'm not sure what you want here.  innerHTML is just the text within the `<text></text>`, `<rect></rect>` does not have any text inside.  It was also assumed that there was only one level of children according to the structure of the SVG, so there would be no more tags within these tags.